examples/backup_vm.py

```diff
@@ -382,16 +382,15 @@ def download_backup(connection, backup, args, incremental=False):
     for disk in backup_disks:
         # During incremental backup, incremental backup may not be available
         # for some of the disks. We need to check the backup mode of the disk.
-        backup_mode = get_disk_backup_mode(connection, disk)
-        has_incremental = backup_mode == types.DiskBackupMode.INCREMENTAL
+        has_incremental = disk.backup_mode == types.DiskBackupMode.INCREMENTAL
```

Comment:
When I tested manually, backup mode was reported correctly, but after applying this change, incremental backup fails:

```
$ ./backup_vm.py -c engine incremental 8e828d5b-d74c-402b-adf7-34e0b9f17fd4 --backup-dir backups --from-checkpoint-uuid c91a3793-7662-4298-b498-ed5ba09823ce
[   0.0 ] Starting incremental backup for VM '8e828d5b-d74c-402b-adf7-34e0b9f17fd4'
[   0.4 ] Waiting until backup 'c9489598-9318-4550-a592-0ee878ade8bd' is ready
[  21.6 ] Created checkpoint 'c422295a-b9e2-4a8f-bd38-22880a8eede0'
[  21.7 ] Incremental backup not available for disk '7719803b-c2f9-4464-9258-c5d3208b9710'
[  21.7 ] Downloading full backup for disk '7719803b-c2f9-4464-9258-c5d3208b9710'
[  21.7 ] Creating backup file 'backups/20220308000448.c422295a-b9e2-4a8f-bd38-22880a8eede0.7719803b-c2f9-4464-9258-c5d3208b9710.full.qcow2'
[  23.3 ] Image transfer '50db53d9-d32c-43e5-afb5-1dd5beb249a1' is ready
[  23.3 ] Finalizing image transfer0 bytes/s                                   
[  25.4 ] Finalizing backup
Traceback (most recent call last):
  File "/home/nsoffer/src/python-ovirt-engine-sdk4/examples/./backup_vm.py", line 543, in <module>
    main()
  File "/home/nsoffer/src/python-ovirt-engine-sdk4/examples/./backup_vm.py", line 145, in main
    args.command(args)
  File "/home/nsoffer/src/python-ovirt-engine-sdk4/examples/./backup_vm.py", line 179, in cmd_incremental
    download_backup(connection, backup, args, incremental=True)
  File "/home/nsoffer/src/python-ovirt-engine-sdk4/examples/./backup_vm.py", line 407, in download_backup
    download_disk(
  File "/home/nsoffer/src/python-ovirt-engine-sdk4/examples/./backup_vm.py", line 452, in download_disk
    client.download(
  File "/usr/lib64/python3.10/site-packages/ovirt_imageio/client/_api.py", line 150, in download
    with _open_http(
  File "/usr/lib64/python3.10/site-packages/ovirt_imageio/client/_api.py", line 514, in _open_http
    return http.open(url, mode, cafile=cafile, secure=secure)
  File "/usr/lib64/python3.10/site-packages/ovirt_imageio/_internal/backends/http.py", line 50, in open
    return Backend(url, **options)
  File "/usr/lib64/python3.10/site-packages/ovirt_imageio/_internal/backends/http.py", line 78, in __init__
    self._connect()
  File "/usr/lib64/python3.10/site-packages/ovirt_imageio/_internal/backends/http.py", line 121, in _connect
    options = self._options()
  File "/usr/lib64/python3.10/site-packages/ovirt_imageio/_internal/backends/http.py", line 487, in _options
    raise self._reraise(res.status, body)
  File "/usr/lib64/python3.10/site-packages/ovirt_imageio/_internal/backends/http.py", line 590, in _reraise
    raise http.Error(status, msg)
ovirt_imageio._internal.http.Error: Server failed to perform the request, check logs
```

Can be also issue with new engine build, need more time to debug this.

Reply:
This was engine regression, works with latest hybrid backup build.